### PR TITLE
Ensure define_hook handles hook kwargs even if it already being defined (closes #36)

### LINKED ISSUE #36: Gossip ignore define_hook kwargs if the hook was registered before
Reproduction code:
```
import gossip

@gossip.register('hook_a')
def foo(**kwargs):
    pass

gossip.define('hook_a', tags=('my', 'tags'), arg_names=('x', 'y'))
gossip.define('hook_b', tags=('my', 'tags'), arg_names=('z', 'y'))

@gossip.register('hook_b')
def bar(**kwargs):
    pass

for hook in gossip.get_all_hooks():
  print(f'Hook: {hook.full_name}, tags: {hook.tags}, arguments: {hook.get_argument_names()}')
```
The output:
```
Hook: hook_a, tags: ('my', 'tags'), arguments: ()
Hook: hook_b, tags: ('my', 'tags'), arguments: ('z', 'y')
```
`hook_b` is as expected, but `hook_a` should add `x` & `y` as arguments

### PR DESCRIPTION
**Pull request by ayalash**

### CODE REVIEW

File: gossip/hooks.py [hunk 1]
```diff
@@ -36,14 +36,34 @@ def __init__(self, group, name, arg_names=None, doc=None, deprecated=False, can_
...
-        self._can_be_muted = can_be_muted
-        self.doc = doc
-        self.deprecated = deprecated
+        self._can_be_muted = None
+        self.doc = None
+        self.deprecated = None
+        self.set_hook_kwargs(arg_names=arg_names, doc=doc, deprecated=deprecated, can_be_muted=can_be_muted)
+
+    def set_hook_kwargs(self, **kwargs):
+        normalized_args = self._normalize_arguments(kwargs.pop('arg_names', None))
```
Comment (vmalloc) [inline gossip/hooks.py]: No check is made for garbage keyword arguments

Comment (vmalloc) [reply gossip/hooks.py]: I would rename this function. Perhaps initialize or something like that

Comment (vmalloc) [reply gossip/hooks.py]: Perhaps configure?

Comment (ayalash) [reply gossip/hooks.py]: Right... missed the check.
I pushed again, please review.

**Comment by vmalloc** [comment]
Thanks!

**Pull request merged**